frontend/js/src/entity-pages/AlbumPage.tsx

```diff
@@ -164,13 +168,16 @@ export default function AlbumPage(props: AlbumPageProps): JSX.Element {
           (track): Listen =>
             popularRecordingToListen(
               merge(track, {
-                artist_name: artistName,
-                artist_mbids:
-                  artist?.artists?.map((ar) => ar.artist_mbid) ?? [],
+                artist_name: track.artists
+                  .map(
+                    (track_artist) =>
+                      `${track_artist.artist_credit_name}${track_artist.join_phrase}`
+                  )
+                  .join(""),
                 caa_id: Number(caa_id) || undefined,
                 caa_release_mbid,
                 recording_name: track.name,
-                release_mbid: caa_release_mbid ?? "",
+                release_mbid: recordings_release_mbid ?? "",
```

Comment:
Would this make sense, just to ensure a cover art fallback?
```suggestion
                release_mbid: recordings_release_mbid ?? caa_release_mbid ?? "",
```